lark/exceptions.py

```diff
@@ -116,44 +118,71 @@ def match_examples(self, parse_fn, examples, token_type_match_fallback=False, us
 
         return candidate[0]
 
+    def _format_terminals(self, terminals):
+        if self._terminals_by_name:
+            ts = []
+            for ter in terminals:
+                ts.append(self._terminals_by_name[ter].user_repr())
```

Comment:
Why not `ts = [t.user_repr() for t in self._terminals_by_name.values()]` ?

Reply:
Because `self.terminals!=self._terminals_by_name`.

Reply:
They are the same length, so how are they different?

Reply:
`terminals` is the collection of names that is passed to `_format_terminals`. This is the expect set. I will rename this.

`_terminals_by_name` is a dictionary.

Reply:
But isn't `set(terminals) == set(_terminals_by_name)`?

Reply:
Awaiting response.